scripts/ci/package-compatibility.mjs

```diff
@@ -0,0 +1,35 @@
+import {publint} from 'publint';
+
+const {messages: atomicMessages} = await publint({
```

Comment:
This is the file where we want to test whether or not are packages are compatible using [publint](https://publint.dev/) and [attw](https://arethetypeswrong.github.io/)

## Publint
Publint should be easy since they have a js library available. My code here should cover most of our case. We still have warnings for now so only fail if errors. We could add atomic-hosted-page, atomic-angular, bueno, auth, headless-react and more if we want but those three are fine for now imo.

## ATTW 
This one should be a bit harder since there is no js library. You only have a cli tool. So you would have to get the output from this https://www.npmjs.com/package/@arethetypeswrong/cli and compare it to what we want to have. Right now we have issues in CJS, it is fine for now and node10 is meant to fail.

Reply:
Types in atomic are meant to be broken. this is what attw should look like: 

atomic-react
"@coveo/atomic-react"

node10: 🟢
node16 (from CJS): 👺 Masquerading as ESM
node16 (from ESM): 🟢 (ESM)
bundler: 🟢

***********************************

"@coveo/atomic-react/commerce"

node10: 💀 Resolution failed
node16 (from CJS): 👺 Masquerading as ESM
node16 (from ESM): 🟢 (ESM)
bundler: 🟢

***********************************

"@coveo/atomic-react/recommendation"

node10: 💀 Resolution failed
node16 (from CJS): 👺 Masquerading as ESM
node16 (from ESM): 🟢 (ESM)
bundler: 🟢

***********************************

atomic
"@coveo/atomic/loader"

node10: 🟢
node16 (from CJS): 🥴 Internal resolution error
node16 (from ESM): 🥴 Internal resolution error
🚭 Unexpected module syntax
bundler: 🟢

***********************************

"@coveo/atomic"

node10: 🟢
node16 (from CJS): ⚠️ ESM (dynamic import only)
🚭 Unexpected module syntax
🥴 Internal resolution error
node16 (from ESM): 🥴 Internal resolution error
bundler: 🟢

***********************************

"@coveo/atomic/themes/*"

node10: (wildcard)
node16 (from CJS): (wildcard)
node16 (from ESM): (wildcard)
bundler: (wildcard)

***********************************

"@coveo/atomic/assets/*"

node10: (wildcard)
node16 (from CJS): (wildcard)
node16 (from ESM): (wildcard)
bundler: (wildcard)

***********************************

"@coveo/atomic/lang/*"

node10: (wildcard)
node16 (from CJS): (wildcard)
node16 (from ESM): (wildcard)
bundler: (wildcard)

***********************************

headless
"@coveo/headless"

node10: 🟢
node16 (from CJS): 👺 Masquerading as ESM
node16 (from ESM): 🟢 (ESM)
bundler: 🟢

***********************************

"@coveo/headless/case-assist"

node10: 💀 Resolution failed
node16 (from CJS): 👺 Masquerading as ESM
node16 (from ESM): 🟢 (ESM)
bundler: 🟢

***********************************

"@coveo/headless/commerce"

node10: 💀 Resolution failed
node16 (from CJS): 👺 Masquerading as ESM
node16 (from ESM): 🟢 (ESM)
bundler: 🟢

***********************************

"@coveo/headless/insight"

node10: 💀 Resolution failed
node16 (from CJS): 👺 Masquerading as ESM
node16 (from ESM): 🟢 (ESM)
bundler: 🟢

***********************************

"@coveo/headless/recommendation"

node10: 💀 Resolution failed
node16 (from CJS): 👺 Masquerading as ESM
node16 (from ESM): 🟢 (ESM)
bundler: 🟢

***********************************

"@coveo/headless/ssr"

node10: 💀 Resolution failed
node16 (from CJS): 👺 Masquerading as ESM
node16 (from ESM): 🟢 (ESM)
bundler: 🟢

***********************************

"@coveo/headless/ssr-commerce"

node10: 💀 Resolution failed
node16 (from CJS): 👺 Masquerading as ESM
node16 (from ESM): 🟢 (ESM)
bundler: 🟢

***********************************

Reply:
After some tinkering around, I decided not to do the attw check.

The attw CLI is not flexible enough for our purposes. While it's possible to specify a set of rules to ignore, it's not possible to specify for instance that you want to ignore "Resolution failed" only in node10.

I tried to output the result of the command in JSON, which we could then parse in order to only raise process.exit(1) when a "valid" error is detected, but the way the CLI is made, it will process.exit(1) when any error is detected, which means I have to wrap the command in a try-catch in order to run it over all of our packages. When I do this, the JSON stdout gets truncated (even if I write it to a file). In a last desperate attempt, I tried to echo the output of the command directly to a file, but that doesn't work either: the JSON gets truncated.

A possible solution would be to output as ascii (which, at least for now, doesn't get truncated because it's short enough), and compare to an expected ascii output... but this would be very brittle and would, IMO, waste more time than it would save.

In short, I don't think it's worth spending any more time trying to integrate an attw scan in our CI. It's a tool that we can use on our own, but it's ill-adapted to CI use.